src/components/EntityPage/EntityScorecardsCard.tsx

```diff
@@ -75,9 +79,41 @@ export const EntityScorecardsCard = ({
 }: EntityScorecardsCardProps) => {
   const classes = useStyles();
 
-  const [sortBy, setSortBy] = useDropdown('Name ↑');
   const [searchQuery, setSearchQuery] = useInput();
 
+  const {
+    compareFn: scorecardCompareFn,
+    loading: loadingScorecardCompareFn,
+    error: scorecardCompareFnError,
+  } = usePartialScorecardCompareFn();
+
+  const customScoresSortMethods: SortMethods<ServiceScorecardScore> =
+    useMemo(() => {
+      if (isUndefined(scorecardCompareFn)) {
+        return {} as SortMethods<ServiceScorecardScore>;
+      }
+
+      return {
+        'Custom ↑': (a: ServiceScorecardScore, b: ServiceScorecardScore) =>
+          scorecardCompareFn(a.scorecard, b.scorecard),
+        'Custom ↓': (a: ServiceScorecardScore, b: ServiceScorecardScore) =>
+          -1 * scorecardCompareFn(a.scorecard, b.scorecard),
+      };
+    }, [scorecardCompareFn]);
+
+  const scorecardScoresSortMethods: SortMethods<ServiceScorecardScore> =
+    useMemo(() => {
+      return {
+        ...customScoresSortMethods,
+        ...defaultSortMethods,
+      };
+    }, [customScoresSortMethods]);
+
+  const [sortBy, setSortBy] = useDropdown(
+    isEmpty(customScoresSortMethods) ? 'Name ↑' : 'Custom ↑',
+    [isEmpty(customScoresSortMethods)],
```

Comment:
I'm kinda confused by this, can you explain?

Reply:
basically want the `useDropdown` initial state to be dependent on `customScoresSortMethods` which is dynamic
if there are no custom sort methods, the default should be on `Name`, otherwise on the `Custom` thing they've defined 

also react didn't seem to like having lambdas in the dependencies (like if i just passed `customScoresSortMethods` as a dep) it always just rerenders every single time, so had to give it a boolean

Reply:
Usually you just don't render the component (in this case, dropdown) until you have all of the info you need to initialize it.